src/IW/Sync/Update.hs

```diff
@@ -1,40 +1,86 @@
+{-# LANGUAGE MultiWayIf #-}
+
 {- | This module contains functions that are used for updating the database.
 It combines functionality from @IW.Sync.Search@ and @IW.Db@ to fetch the latest
 data and insert it into the database.
 -}
 
 module IW.Sync.Update
-       ( fetchAndUpsertRepos
+       ( syncRepos
        ) where
 
 import Control.Monad.IO.Unlift (MonadUnliftIO)
+import Data.Time (Day (..), addDays)
 import UnliftIO.Async (mapConcurrently_)
 
 import IW.App (WithError)
 import IW.Core.Repo (Repo (..))
 import IW.Db (WithDb, upsertRepos, updateRepoCategories)
 import IW.Effects.Cabal (MonadCabal (..), getCabalCategories)
-import IW.Sync.Search (fetchAllHaskellRepos, fromGitHubRepo)
+import IW.Sync.Search (fetchHaskellReposByDate, fromGitHubRepo)
+
+
+-- | This function fetches repos from the GitHub API within a specified date range,
+-- parses their @.cabal@ files, and upserts them into the database.
+syncRepos
+    :: forall env m.
+       ( MonadCabal m
+       , MonadUnliftIO m
+       , WithDb env m
+       , WithLog env m
+       , WithError m
+       )
+    => Day     -- ^ Oldest day
+    -> Day     -- ^ Most recent day
+    -> Integer -- ^ Interval of days before most recent day
+    -> Int     -- ^ Page
+    -> m ()
+syncRepos oldest recent interval page =
+    if recent == oldest then
+        log I $ "Oldest day reached"
+    else
+        do
+            resCount <- syncReposByDate intervalStart recent page
+            if resCount < 100 then
+                syncRepos oldest nextRecent interval 1
+            else
```

Comment:
According to our style guide this should be:

```haskell
if ...
then syncRepos
else syncRepos
```

Reply:
The style guide also mentions this style for `do` blocks:
```
    if null answer
        then pure (head choices)
        else pure answer
```

Reply:
@rashadg1030 Yeah, that's true. I think both are fine. Yours is preferrable to be consistent with the style guide